notebooks/05-operations/01-druid-logs-configuration.ipynb

```diff
@@ -0,0 +1,511 @@
+    "    -f historical.log\n",
+    "```\n",
+    "\n",
+    "Open a new terminal window and run this command to amend the base logging level for all Druid processes:\n",
+    "\n",
+    "```bash\n",
+    "sed -i '' 's/Root level=\"info\"/Root level=\"debug\"/' \\\n",
+    "  ~/learn-druid-local/apache-druid/conf/druid/auto/_common/log4j2.xml\n",
+    "```\n",
+    "\n",
+    "Within a short period of time you will see many more log messages appearing.\n",
+    "\n",
+    "Revert the logging level to INFO before proceeding.\n",
+    "\n",
+    "```bash\n",
+    "sed -i '' 's/Root level=\"debug\"/Root level=\"info\"/' \\\n",
+    "  ~/learn-druid-local/apache-druid/conf/druid/auto/_common/log4j2.xml\n",
+    "```\n",
+    "\n",
+    "Leave `multitail` running and your second terminal open."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7e0a46b8-77f4-4792-9484-9acba0407b3d",
+   "metadata": {},
+   "source": [
+    "## Examples\n",
+    "\n",
+    "Two recommended approaches for working with log files are:\n",
+    "\n",
+    "1. Search individual log files for indications of a problem, such as WARN, ERROR, and FATAL, and of Java exceptions in stack traces (e.g. `java.net.ConnectionException: Connection refused`.\n",
+    "2. Read log files more like a novel, starting at the beginning of the file - or some known intermediate point - and then follow the story logically.\n",
+    "\n",
+    "In approach 1, it's possible to work back through the history, using the time field as a key. You may notice large time gaps, or use filtering to remove any events from threads or classes that you know are not relevant to the error.\n",
+    "\n",
+    "Approach 2 requires more time, but helps to solve more complex problems. It's also an important learning aid when delving deep into how Druid's processes collaborate to realise services.\n",
+    "\n",
+    "In this section you'll find examples of log files being useful in diagnosing issues with Druid."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f9137fc3-9f53-4887-a2b2-b8cabb061cc5",
+   "metadata": {},
+   "source": [
+    "### Example: failure in Apache Zookeeper\n",
+    "\n",
+    "Druid's process rely on [Apache Zookeeper](https://zookeeper.apache.org/) for [inter-process communication and configuration](https://druid.apache.org/docs/latest/dependencies/zookeeper.html). Run the following command to simulate a failure in Zookeeper.\n",
+    "\n",
+    "Run the following command in your terminal window. The first `kill` prevents Zookeeper from being recovered by the `learn-druid` script's `supervisor`, and the second `kill` stops the Zookeeper process.\n",
+    "\n",
+    "```bash\n",
+    "kill -STOP $(ps -ef | grep 'perl' | awk 'NF{print $2}' | head -n 1)\n",
+    "kill $(ps -ef | grep 'zoo' | awk 'NF{print $2}' | head -n 1)\n",
+    "```\n",
+    "\n",
+    "In your `multitail` window you will now see multiple `INFO` messages relating to Zookeeper connection issues.\n",
```

Comment:
perhaps include some of the messages you are referring to, while there are a bunch of INFO messages about opening sockets, it is not evident that these are Zookeeper related.
```
2-13T18:48:19,210 INFO [main-SendThread(localhost:2181)] org.apache..ClientCnxn - Opening socket connection t d: localhost/127.0.0.1:2181: Connection refused
o server localhost/127.0.0.1:2181. Will not attempt to authenticate using SASL (unknown error)                      2024-02-13T18:48:24,735 INFO [main-SendThread(localhost:2181)] org.apache.zookeeper.ClientCnxn - Opening socket connection ...
                             
```